components/org.wso2.carbon.extension.identity.sso.jwt/src/main/java/org/wso2/carbon/identity/sso/jwt/processor/JWTInboundRequestProcessor.java

```diff
@@ -0,0 +1,669 @@
+                // Handle logout request.
+                if (log.isDebugEnabled()) {
+                    log.debug("Handling logout request.");
+                }
+                return buildResponseForFrameworkLogout(messageContext);
+            } else if (StringUtils.isNotBlank(sessionId)) {
+                // A session exists - response coming from the framework after authentication or after logged out.
+                return handleFrameworkResponse(messageContext, identityRequest, respBuilder);
+            } else {
+                // No session exists - send the request to the identity framework.
+                if (log.isDebugEnabled()) {
+                    log.debug("Session not exists. Sending request to identity framework for authentication.");
+                }
+                return buildResponseForFrameworkLogin(messageContext);
+            }
+        } else {
+            String msg = "A Service Provider with the Relying Party '" + neutralize(this.relyingParty) + "' is not " +
+                    "registered. Service Provider should be registered in advance.";
+            log.error(msg, new JWTIdentityException(msg));
+            return JWTInboundUtil.sendToRetryPage(JWTInboundConstants.ErrorMessages.MISCONFIGURATION_STATUS,
+                    JWTInboundConstants.ErrorMessages.MISCONFIGURATION_MESSAGE);
+        }
+    }
+
+    /**
+     * Validates the endpoint URL
+     *
+     * @param identityRequest The identity request
+     * @return
+     */
+    private boolean validateEndpointUrl(IdentityRequest identityRequest) {
+
+        this.endpointUrl = getPropertyValue(identityRequest, JWTInboundConstants.SPBasedConfigs.SITE_API_URL);
+        if (StringUtils.isNotBlank(this.endpointUrl)) {
+            if (log.isDebugEnabled()) {
+                log.debug("Setting the endpoint URL: " + neutralize(this.endpointUrl));
+            }
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Validates the JWS algorithm
+     *
+     * @param identityRequest The identity request
+     * @return
+     */
+    private boolean validateJwsAlgorithm(IdentityRequest identityRequest) {
+
+        setJWSAlgorithm(getPropertyValue(identityRequest, JWTInboundConstants.SPBasedConfigs.JWS_ALGORITHM));
+        if (this.jwsAlgorithm != null) {
+            if (log.isDebugEnabled()) {
+                log.debug("Setting the JWS Algorithm");
+            }
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Setting the redirect URL parameter and validating against the regex configured in the SP configuration.
+     *
+     * @param identityRequest The identity request
+     * @return True if the redirect URL is valid or not provided. False if the validation failed against the regex.
+     */
+    private boolean handleRedirectUrl(IdentityRequest identityRequest) {
+
+        // Set Redirect URL
```

Comment:
Add full stop at the end check other places as well

Reply:
Fixed